src/twisted/internet/test/test_inlinecb.py

```diff
@@ -3,22 +3,332 @@
 # See LICENSE for details.
 
 """
-Tests for L{twisted.internet.defer.inlineCallbacks}.
-
-Some tests for inlineCallbacks are defined in L{twisted.test.test_defgen} as
-well.
+Tests for L{twisted.internet.inlineCallbacks}.
 """
 
+import traceback
+import unittest as pyunit
+import weakref
+from typing import Any, Generator, List, Set, Union
 
+from twisted.internet import reactor, task
 from twisted.internet.defer import (
     CancelledError,
     Deferred,
+    Failure,
+    fail,
     inlineCallbacks,
     returnValue,
+    succeed,
 )
+from twisted.python.compat import _PYPY
 from twisted.trial.unittest import SynchronousTestCase, TestCase
 
 
+def getThing():
+    d = Deferred()
+    reactor.callLater(0, d.callback, "hi")
+    return d
+
+
+def getDivisionFailure(msg: Union[str, None] = None) -> Failure:
+    """
+    Make a L{Failure} of a divide-by-zero error.
+    """
+    try:
+        raise ZeroDivisionError(msg)
+    except BaseException:
+        f = Failure()
+    return f
+
+
+class TerminalException(Exception):
```

Comment:
Can you please add a docstring explaining the purpose of this exception.

The descring should help other developers if it's ok to re-use this exception in other tests, or is best to create another exception.

Reply:
Done. It's just a separate exception type that is used within this file. There's nothing unique in it.